flang/lib/Lower/Runtime.cpp

```diff
@@ -243,7 +246,60 @@ void Fortran::lower::genDateAndTime(Fortran::lower::FirOpBuilder &builder,
   llvm::SmallVector<mlir::Value, 8> args{dateBuffer, timeBuffer, zoneBuffer,
                                          dateLen,    timeLen,    zoneLen};
   llvm::SmallVector<mlir::Value, 8> operands;
-  for (auto [fst,snd] : llvm::zip(args, callee.getType().getInputs()))
-    operands.emplace_back(builder.convertWithSemantics(loc, snd, fst));
+  for (auto [fst, snd] : llvm::zip(args, callee.getType().getInputs()))
+    operands.emplace_back(builder.createConvert(loc, snd, fst));
   builder.create<fir::CallOp>(loc, callee, operands);
 }
+
+void Fortran::lower::genAllocatableInitIntrinsic(
+    FirOpBuilder &builder, mlir::Location loc, mlir::Value boxAddress,
+    mlir::Value typeCategory, mlir::Value kind, mlir::Value rank,
+    mlir::Value corank) {
+  auto callee =
+      genRuntimeFunction<mkRTKey(AllocatableInitIntrinsic)>(loc, builder);
+  llvm::SmallVector<mlir::Value, 5> args{boxAddress, typeCategory, kind, rank,
+                                         corank};
```

Comment:
So the allocation runtime is going to be co-array aware?

Nit: I'd use `var = { ... }` to make it clear this is an initializer list and not a constructor.

Reply:
Well, it has the corank in its interface, but I am not sure there is a detail plan for co-array yet.
I would imagine a different entry point might make sense at some point.